quil-rs/src/expression/simplification/by_hand.rs

```diff
@@ -0,0 +1,830 @@
+
+        // Adding with zero
+        (Expression::Number(x), InfixOperator::Plus, right) if is_zero(x) => right,
+        (left, InfixOperator::Plus, Expression::Number(x)) if is_zero(x) => left,
+        // Adding numbers or π
+        (Expression::Number(x), InfixOperator::Plus, Expression::Number(y)) => {
+            Expression::Number(x + y)
+        }
+        (Expression::Number(x), InfixOperator::Plus, Expression::PiConstant) => {
+            Expression::Number(x + PI)
+        }
+        (Expression::PiConstant, InfixOperator::Plus, Expression::Number(y)) => {
+            Expression::Number(PI + y)
+        }
+        (Expression::PiConstant, InfixOperator::Plus, Expression::PiConstant) => {
+            Expression::Number(2.0 * PI)
+        }
+
+        // Subtracting with zero
+        (Expression::Number(x), InfixOperator::Minus, right) if is_zero(x) => {
+            simplify_prefix(PrefixOperator::Minus, &right, limit.saturating_sub(2))
+        }
+        (left, InfixOperator::Minus, Expression::Number(y)) if is_zero(y) => left,
+        // Subtracting self
+        (left, InfixOperator::Minus, right) if left == right => Expression::Number(ZERO),
+        // Subtracting numbers or π (π - π already covered)
+        (Expression::Number(x), InfixOperator::Minus, Expression::Number(y)) => {
+            Expression::Number(x - y)
+        }
+        (Expression::Number(x), InfixOperator::Minus, Expression::PiConstant) => {
+            Expression::Number(x - PI)
+        }
+        (Expression::PiConstant, InfixOperator::Minus, Expression::Number(y)) => {
+            Expression::Number(PI - y)
+        }
+
+        // Multiplication and Division
+
+        // Multiplication with zero
+        (Expression::Number(x), InfixOperator::Star, _) if is_zero(x) => Expression::Number(ZERO),
+        (_, InfixOperator::Star, Expression::Number(y)) if is_zero(y) => Expression::Number(ZERO),
+        // Multiplication with one
+        (Expression::Number(x), InfixOperator::Star, right) if is_one(x) => right,
+        (left, InfixOperator::Star, Expression::Number(y)) if is_one(y) => left,
+        // Multiplying with numbers or π
+        (Expression::Number(x), InfixOperator::Star, Expression::Number(y)) => {
+            Expression::Number(x * y)
+        }
+        (Expression::Number(x), InfixOperator::Star, Expression::PiConstant) => {
+            Expression::Number(x * PI)
+        }
+        (Expression::PiConstant, InfixOperator::Star, Expression::Number(y)) => {
+            Expression::Number(PI * y)
+        }
+        (Expression::PiConstant, InfixOperator::Star, Expression::PiConstant) => {
+            Expression::Number(PI * PI)
+        }
+
+        // Division with zero
+        (Expression::Number(x), InfixOperator::Slash, _) if is_zero(x) => Expression::Number(ZERO),
+        (_, InfixOperator::Slash, Expression::Number(y)) if is_zero(y) => {
+            Expression::Number(real!(f64::NAN)) // TODO Is this OK?
```

Comment:
TODO

Reply:
Yes, thanks! I think we can just remove the comment, but let me know if you think otherwise.